gpu_tests/test_hub_utils.py

```diff
@@ -0,0 +1,187 @@
+        # 'batch_size': None,
+        # 'batch_size_valid': None
+    }
+
+    # Need to enable this context otherwise fsdp_wrap is a no op. See fairscale.nn.wrap
+    with fsdp_enable_wrap(cfg.distributed_training, use_sharded_state=True):
+        x = hub_utils.from_pretrained(
+            model_path,
+            "reshard.pt",
+            **kwargs,
+        )
+
+    generator = GeneratorHubInterface(
+        x["args"],
+        x["task"],
+        x["models"],
+        moe_disable_padding=False,
+        skip_prepare_for_inference=True,
+    )
+
+    override_task(generator)
+    generator = generator.cuda()
+
+    tokenized_sentences = [torch.tensor(PROMPT)]
+    hypothesis = generator.generate(
+        tokenized_sentences=tokenized_sentences,
+        score_reference=True,
+        batch_size=1,
+        compute_vocab_dist=False,
+    )
+
+    # Destroy torch distributed process groups. This needs to be executed in each spawned process
+    # https://github.com/pytorch/pytorch/issues/48203
+    dist.destroy_process_group()
+    return hypothesis
+
+
+# TEST FUNCTIONS #
+def test_generator_interface(data_regression, ndarrays_regression):
+    model_path = os.path.join(os.path.dirname(__file__), "125m")
+    cfg = create_generation_config_with_defaults(
+        model_path, ddp_backend="fully_sharded"
+    )
+
+    overall_generation = distributed_utils.call_main(
+        cfg, generate_using_generator_interface
+    )
+    # We can potentially pass a list of inputs to the generate function.
+    # Here, we look at the first (and only) generation
+    [generation_for_prompt] = overall_generation
+    # We use best_of = 1, so we get only one beam search result
+    [generated_beam] = generation_for_prompt
+
+    ndarray_data = {
+        "token_scores": np.array(
+            [
+                np.nan if elem is None else elem
+                for elem in generated_beam["token_scores"]
+            ]
+        )
+    }
+    generated_beam.pop("token_scores")
+
+    ndarrays_regression.check(ndarray_data, default_tolerance=dict(atol=1e-2))
+    data_regression.check(generated_beam)
+
+
+def test_generator_hub_interface(data_regression, ndarrays_regression):
+    model_path = os.path.join(os.path.dirname(__file__), "125m")
+    cfg = create_generation_config_with_defaults(
+        model_path, ddp_backend="fully_sharded"
+    )
+
+    overall_generation = distributed_utils.call_main(
+        cfg, generate_using_generator_hub_interface, model_path=model_path
+    )
+    # We can potentially pass a list of inputs to the generate function.
+    # Here, we look at the first (and only) generation
+    [generation_for_prompt] = overall_generation
+    # We use best_of = 1, so we get only one beam search result
+    [generated_beam] = generation_for_prompt
+
+    ndarray_data = {
+        key: value.cpu()
+        for key, value in generated_beam.items()
+        if key in ["tokens", "score", "positional_scores"]
```

Comment:
The tokens here I believe are the indexes, can we also test the decoded string, as decode is part of generator_hub_interface.